appinsights/client.go

```diff
@@ -2,123 +2,129 @@ package appinsights
+	item.Context.InstrumentationKey = iKey
 
-		for tagkey, tagval := range clientContext.tags {
-			if itemContext.tags[tagkey] == "" {
-				itemContext.tags[tagkey] = tagval
-			}
+	for tagkey, tagval := range tc.Context.Tags {
+		if item.Context.Tags[tagkey] == "" {
+			item.Context.Tags[tagkey] = tagval
 		}
-
-		tc.channel.Send(item)
 	}
-}
-
-func (tc *telemetryClient) TrackEvent(name string) {
-	item := NewEventTelemetry(name)
-	tc.TrackEventTelemetry(item)
-}
-
-func (tc *telemetryClient) TrackEventTelemetry(event *EventTelemetry) {
-	var item Telemetry
-	item = event
-
-	tc.Track(item)
-}
-
-func (tc *telemetryClient) TrackMetric(name string, value float32) {
-	item := NewMetricTelemetry(name, value)
-	tc.TrackMetricTelemetry(item)
-}
-
-func (tc *telemetryClient) TrackMetricTelemetry(metric *MetricTelemetry) {
-	var item Telemetry
-	item = metric
-
-	tc.Track(item)
-}
-
-func (tc *telemetryClient) TrackTrace(message string) {
-	item := NewTraceTelemetry(message, Information)
-	tc.TrackTraceTelemetry(item)
-}
-
-func (tc *telemetryClient) TrackTraceTelemetry(trace *TraceTelemetry) {
-	var item Telemetry
-	item = trace
-
-	tc.Track(item)
-}
-
-func (tc *telemetryClient) TrackRequest(name string, timestamp time.Time, duration time.Duration, responseCode string, success bool) {
-	item := NewRequestTelemetry(name, timestamp, duration, responseCode, success)
-	tc.TrackRequestTelemetry(item)
-}
-
-func (tc *telemetryClient) TrackRequestTelemetry(request *RequestTelemetry) {
-	var item Telemetry
-	item = request
 
-	tc.Track(item)
+	tc.Channel.Send(item)
+}
+
+func (tc *TelemetryClient) TrackEvent(name string) {
+	tc.Track(NewEventTelemetry(name))
+}
+
+func (tc *TelemetryClient) TrackMetric(name string, value float32) {
+	tc.Track(NewMetricTelemetry(name, value))
+}
+
+func (tc *TelemetryClient) TrackTrace(
+	level SeverityLevel,
+	message string,
+	properties map[string]string,
+	alterTelementryContext func(*TelemetryContext)) {
+
+	tc.Track(NewTraceTelemetry(
+		message,
+		properties,
+		level,
+		alterTelementryContext))
+}
+
+func (tc *TelemetryClient) TrackRequest(
+	id string,
+	name string,
+	timestamp time.Time,
+	duration time.Duration,
+	httpMethod string,
+	url string,
+	responseCode string,
+	success bool,
+	properties map[string]string,
+	alterTelementryContext func(*TelemetryContext)) {
+
+	tc.Track(NewRequestTelemetry(
+		id,
+		name,
+		timestamp,
+		duration,
+		httpMethod,
+		url,
+		responseCode,
+		success,
+		properties,
+		alterTelementryContext))
+}
+
+func (tc *TelemetryClient) TrackRemoteDependency(
+	id string,
+	name string,
+	resultCode int,
+	commandName string,
+	kind DataPointType,
+	duration time.Duration,
+	count int,
+	min float32,
+	max float32,
+	stdDev float32,
+	theType string,
+	dependencyKind DependencyKind,
+	success bool,
+	async bool,
+	dependencySource DependencySourceType,
+	properties map[string]string,
+	alterTelementryContext func(*TelemetryContext)) {
```

Comment:
I would also add: here are [the functions used in .Net](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Core/Managed/Shared/TelemetryClient.cs#L276) to point out the signatures.